kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/BigQuerySinkTask.java

```diff
@@ -110,7 +112,19 @@ private void updateOffsets(Map<TopicPartition, OffsetAndMetadata> offsets) {
 
   private PartitionedTableId getRecordTable(SinkRecord record) {
     TableId baseTableId = topicsToBaseTableIds.get(record.topic());
-    return new PartitionedTableId.Builder(baseTableId).setDayPartitionForNow().build();
+
+    PartitionedTableId.Builder builder = new PartitionedTableId.Builder(baseTableId);
+    if (useMessageTimeDatePartitioning) {
+      if (record.timestampType() == TimestampType.NO_TIMESTAMP_TYPE) {
+        throw new ConnectException("Message has no timestamp type, cannot use message timestamp to partition.");
```

Comment:
When would a record not have a timestamp type? Is it possible we want to simply warn rather than throw an exception?

Reply:
`NO_TIMESTAMP_TYPE` can happen in SinkRecords for older versions of Kafka before timestamps were supported. When the record has no timestamp, the timestamp field will be null. I have this in here as a safety check, timestamp support was added by [KIP-32](https://cwiki.apache.org/confluence/display/KAFKA/KIP-32+-+Add+timestamps+to+Kafka+message) for 0.10.0.0+